test/ooni/dns_injection.cpp

```diff
@@ -5,11 +5,89 @@
 #define CATCH_CONFIG_MAIN
 #include "src/ext/Catch/single_include/catch.hpp"
 
+#include "src/ooni/collector_client_impl.hpp"
 #include "src/ooni/dns_injection_impl.hpp"
 #include <measurement_kit/common.hpp>
 
-using namespace mk;
 using namespace mk::ooni;
+using namespace mk::report;
+using namespace mk;
+
+TEST_CASE("Check dns-injection output for injected entry") {
+    loop_with_initial_event([=]() {
+        dns_injection("8.8.8.8:53", "nexa.polito.it", [=](Error err,
+                                                          Var<Entry> entry) {
+            // We have called the DNS injection test with an actually
```

Comment:
The main advantage of this new code is that, as shown here, we can call and test directly the code for running this specific test (i.e. we can test a smaller bit of code and go towards precise unit testing)